tests/ResultTests.cpp

```diff
@@ -575,6 +577,31 @@ TEST_CASE("ResultTests::AutomaticOriginationOnFailure", "[result]")
     }();
     REQUIRE(S_FALSE == GetRestrictedErrorInfo(&restrictedErrorInformation));
 }
+
+TEST_CASE("ResultTests::OriginatedWithMessagePreserved", "[result]")
+{
+#ifdef WIL_ENABLE_EXCEPTIONS
+    try
+    {
+        THROW_HR_MSG(E_FAIL, "Puppies not allowed");

```

Comment:
Please add a test case for `std::exception` too.  That is the impetus for this entire change so it would be nice to have explicit test coverage.  `wil::ResultException` already has decent logging even before this change.

Reply:
Drat, I was too quick on the draw. I'll do that, @jlaanstra.

Reply:
PS - manual testing shows the error comes through.